### ISSUE TITLE
warn when attempting to manipulate a macro that appears to not exist

### ISSUE THREAD

**Issue by mcandre** [mcandre/unmake#130]
Things that would trigger this warning:

* Attempts to append (`:=`) or conditionally assign (`?=`) macros whose names are missing.
* Macro values and expansions that referencing missing macro names, in general expressions and other macro expansions.

Depends on #85.

We also need to identify predefined POSIX macro names such as `MAKE` and so on.

We may also need to provide a comment syntax to indicate implicit includes, such as `sys.mk` files for b(sd)make.

**Comment by mcandre** [comment]
#wontfix
